File: botasaurus/decorators_utils.py
import os

def relative_path(path, goback=0):
    levels = [".."] * (goback + -1)
    return os.path.abspath(os.path.join(os.getcwd(), *levels, path.strip()))

def create_directory_if_not_exists(passed_path):
    dir_path = relative_path(passed_path)
    try:
        if not os.path.exists(dir_path):
            os.makedirs(dir_path)
    except FileExistsError:
        pass

File: js/botasaurus-js/src/decorators-utils.ts
import path from 'path'
import fs from 'fs'
import { getTargetDirectoryPath } from './paths'

function relativePath(filePath:string, goback = 0) {
    const levels = Array(goback ).fill('..');
    return path.resolve(getTargetDirectoryPath(), ...levels, filePath.trim());
}

function createDirectoryIfNotExists(passedPath:string) {
    const dirPath = relativePath(passedPath);

    if (!fs.existsSync(dirPath)) {
        fs.mkdirSync(dirPath, { recursive: true });
    }
}


let outputCheckDone = false;

function createOutputDirectoryIfNotExists() {
    if (!outputCheckDone) {
        outputCheckDone = true;
        createDirectoryIfNotExists('output/');
    }
}

export  {
    relativePath,
    createDirectoryIfNotExists,
    createOutputDirectoryIfNotExists
};